### PR TITLE
feat(route53): replace existing record sets

### PR DESCRIPTION
**Pull request by jogold**
Add a `deleteExisting` prop to `RecordSet` to delete an existing record
set before deploying the new one. This is useful if you want to minimize
downtime and avoid "manual" actions while deploying a stack with a
record set that already exists. This is typically the case for record
sets that are not already "owned" by CloudFormation or "owned" by
another stack or construct that is going to be deleted (migration).

----

### All Submissions:

* [x] Have you followed the guidelines in our [Contributing guide?](https://github.com/aws/aws-cdk/blob/master/CONTRIBUTING.md)

### Adding new Unconventional Dependencies:

* [ ] This PR adds new unconventional dependencies following the process described [here](https://github.com/aws/aws-cdk/blob/master/CONTRIBUTING.md/#adding-new-unconventional-dependencies)

### New Features

* [x] Have you added the new feature to an [integration test](https://github.com/aws/aws-cdk/blob/master/INTEGRATION_TESTS.md)?
	* [x] Did you use `yarn integ` to deploy the infrastructure and generate the snapshot (i.e. `yarn integ` without `--dry-run`)?

*By submitting this pull request, I confirm that my contribution is made under the terms of the Apache-2.0 license*

### CODE REVIEW

File: packages/@aws-cdk/aws-route53/README.md [hunk 1]
```diff
@@ -132,6 +132,26 @@ Constructs are available for A, AAAA, CAA, CNAME, MX, NS, SRV and TXT records.
...
+are not already "owned" by CloudFormation or "owned" by another stack or construct that is
+going to be deleted (migration).
+
+```ts
+declare const myZone: route53.HostedZone;
+
+new route53.ARecord(this, 'ARecord', {
+  zone: myZone,
+  target: route53.RecordTarget.fromIpAddresses('1.2.3.4', '5.6.7.8'),
+  deleteExisting: true,
```
Comment (jogold) [inline packages/@aws-cdk/aws-route53/README.md:149]: In the beginning I thought  that `upsert` would have been a good name for this prop, but it's not a real UPSERT operation?

Comment (comcalvi) [reply packages/@aws-cdk/aws-route53/README.md:149]: The existing name is definitely better.

File: packages/@aws-cdk/aws-route53/README.md [hunk 2]
```diff
@@ -171,7 +191,7 @@ new route53.CrossAccountZoneDelegationRecord(this, 'delegate', {
 
 ## Imports
 
-If you don't know the ID of the Hosted Zone to import, you can use the 
+If you don't know the ID of the Hosted Zone to import, you can use the
```
Comment (comcalvi) [inline packages/@aws-cdk/aws-route53/README.md:194]: Good catch!

**Review by comcalvi — APPROVED** [review]
Thanks for the PR!

**Pull request merged**